pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -359,17 +359,38 @@ func TestPriorityQueue_Delete(t *testing.T) {
 }
 
 func TestPriorityQueue_MoveAllToActiveOrBackoffQueue(t *testing.T) {
-	q := NewPriorityQueue(newDefaultQueueSort())
+	c := clock.NewFakeClock(time.Now())
+	q := NewPriorityQueue(newDefaultQueueSort(), WithClock(c))
 	q.Add(&medPriorityPod)
+	q.MoveAllToActiveOrBackoffQueue("test")
+	// Make sure the move request is happened first.
+	c.Step(time.Second)
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&unschedulablePod), q.SchedulingCycle())
 	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&highPriorityPod), q.SchedulingCycle())
-	q.MoveAllToActiveOrBackoffQueue("test")
 	if q.activeQ.Len() != 1 {
-		t.Error("Expected 1 item to be in activeQ")
+		t.Errorf("Expected 1 item to be in activeQ, but got: %v", q.activeQ.Len())
 	}
 	if q.podBackoffQ.Len() != 2 {
-		t.Error("Expected 2 items to be in podBackoffQ")
+		t.Errorf("Expected 2 items to be in podBackoffQ, but got: %v", q.podBackoffQ.Len())
 	}
+
+	// pop out the pods in the backoffQ.
+	q.podBackoffQ.Pop()
+	q.podBackoffQ.Pop()
+
+	q.schedulingCycle++
+	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&unschedulablePod), q.SchedulingCycle())
+	q.AddUnschedulableIfNotPresent(q.newQueuedPodInfo(&highPriorityPod), q.SchedulingCycle())
```

Comment:
so they are going to unschedulable queue here? this is different with line 368 an 369, so it is better to have assert about it.

Reply:
yes.

Reply:
Done